hawc/apps/study/serializers.py

```diff
@@ -29,6 +31,38 @@ class Meta:
         )
 
 
+class CreateStudySerializer(SimpleStudySerializer):
+    def validate(self, data):
+
+        ref_id = self.initial_data.get("reference_id", -1)
+
+        try:
+            reference = Reference.objects.get(id=ref_id)
+        except ValueError:
+            raise serializers.ValidationError("Reference ID must be a number.")
+        except ObjectDoesNotExist:
+            raise serializers.ValidationError(f"Reference ID does not exist.")
+        if models.Study.objects.filter(reference_ptr=ref_id).exists():
+            raise serializers.ValidationError(f"Reference ID {ref_id} already linked with a study.")
+        if not reference.assessment.user_can_edit_object(self.context["request"].user):
+            raise exceptions.PermissionDenied("Invalid permission to edit assessment.")
+        return super().validate(data)
+
+    def create(self, validated_data):
+        reference = Reference.objects.get(id=self.initial_data.get("reference_id"))
+        validated_data["assessment"] = reference.assessment.id
+
+        return models.Study.save_new_from_reference(reference, validated_data)
+
+    class Meta:
```

Comment:
do we need meta, or can it just be inherited from simple?  In fact, can we just add these methods to SmipleStudySerializer and remove this subclass?

Reply:
The meta is slightly different; the CreateStudySerializer also excludes assessment, since we figure that out from the given reference id. Also the views that use the SimpleStudySerializer do not pass in a reference id, so the validation would be different. I'm sure I can combine them and just test on what keys are in the initial data, but it would lead to one more complicated serializer instead of two less complicated serializers. Would you rather that?

Reply:
I combinded the serializers, let me know if you like the looks of it!

Reply:
LGTM